tidy3d/components/scene.py

```diff
@@ -1160,8 +1209,18 @@ def _pcolormesh_shape_custom_medium_structure_eps(
             normal_axis: [normal_position],
         }
         coord_shape = Coords(**coord_dict)
-        # interpolate permittivity and take the average over components
-        eps_shape = np.mean(medium.eps_diagonal_on_grid(frequency=freq, coords=coord_shape), axis=0)
+
+        # interpolate permittivity and pick the component to plot
+        eps_shape = medium.eps_diagonal_on_grid(frequency=freq, coords=coord_shape)
+        if component == "xx":
```

Comment:
similar comment to above. should we just generalize tis to do the the `mean` using the off diagonal components too? or in this case will it create issues if the shape is not the same between eg. x and y permittivity grids?